test/e2e/run_volume_test.go

```diff
@@ -162,4 +162,26 @@ var _ = Describe("Podman run with volumes", func() {
 		Expect(session.OutputToString()).To(ContainSubstring("/testvol1"))
 		Expect(session.OutputToString()).To(ContainSubstring("/testvol2"))
 	})
+
+	It("podman run with volumes and suid/dev/exec options", func() {
+		mountPath := filepath.Join(podmanTest.TempDir, "secrets")
+		os.Mkdir(mountPath, 0755)
+		session := podmanTest.Podman([]string{"run", "--rm", "-v", fmt.Sprintf("%s:/run/test:suid,dev,exec", mountPath), ALPINE, "grep", "/run/test", "/proc/self/mountinfo"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		found, matches := session.GrepString("/run/test")
+		Expect(found).Should(BeTrue())
+		Expect(matches[0]).To(Not(ContainSubstring("noexec")))
+		Expect(matches[0]).To(Not(ContainSubstring("nodev")))
+		Expect(matches[0]).To(Not(ContainSubstring("nosuid")))
+
+		session = podmanTest.Podman([]string{"run", "--rm", "--tmpfs", "/run/test:suid,dev,exec", ALPINE, "grep", "/run/test", "/proc/self/mountinfo"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		found, matches = session.GrepString("/run/test")
+		Expect(found).Should(BeTrue())
+		Expect(matches[0]).To(Not(ContainSubstring("noexec")))
+		Expect(matches[0]).To(Not(ContainSubstring("nodev")))
+		Expect(matches[0]).To(Not(ContainSubstring("nosuid")))
+	})
```

Comment:
Can we add an additional block box test?

A trivial one could be an executable `test.sh`:
```
#!/bin/sh
echo "Hello World!"
```

And then mount & execute it once with `exec` and grep for `Hello World!` and then mount & execute it with `noexec` and grep for ` Permission denied`.

Reply:
Done - I just mounted in `/bin` from the host, but it should be a sufficient test.